### PR TITLE
Day1 - UI creation using jetpack compose

### PR DESCRIPTION
**Pull request by sachinsinghd11**
# Pull Request

## Description

This PR includes implementation of UI screens. There are 2 screens one is RecipeList and other is RecipeDetails

## Checklist

- [ Yes ] My project includes a `README.md` file with a brief overview of the project, how to set it up, and any relevant information.
- [ Yes ] I have added a screenshot or screen recording showing the functionality of the app.
- [ Yes ] I have tested the project and confirmed it works as expected.

## Screenshots

1) Recipe List screen - 
Recipe_list_screen

Recipe_list_screen

2) Recipe Detail Screen - 

Recipe_detail_screen1

Recipe_detail_screen2

## Project README

Please ensure that your project's `README.md` is detailed and includes:

- A brief introduction to the project.
- Installation steps.
- Key features.
- Any additional instructions or information.

## Additional Notes

Is there anything else the maintainers should know about your contribution?

---

Thank you for your contribution!

### CODE REVIEW

File: README.md
Comment (anandwana001) [inline README.md:1]: all files should be under your directory, not in parent

File: SachinSinghDighan/RecipeRoulette/app/src/main/java/com/example/reciperoulette/presentation/recipelist/MainActivity.kt [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+import com.example.reciperoulette.R
+import com.example.reciperoulette.presentation.recipedetail.RecipeDetailActivity
+import com.example.reciperoulette.presentation.recipelist.ui.theme.RecipeRouletteTheme
+
+class MainActivity : ComponentActivity() {
+    @SuppressLint("UnusedMaterial3ScaffoldPaddingParameter")
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContent {
+            Scaffold(topBar = { AppBar("RecipeList") }) {
```
Comment (anandwana001) [inline SachinSinghDighan/RecipeRoulette/app/src/main/java/com/example/reciperoulette/presentation/recipelist/MainActivity.kt]: why hardcode string

File: SachinSinghDighan/RecipeRoulette/app/src/main/java/com/example/reciperoulette/presentation/recipelist/MainActivity.kt [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.example.make_food.ui.commonui.AppBar
+import com.example.make_food.ui.commonui.ScreenContentText
+import com.example.reciperoulette.R
+import com.example.reciperoulette.presentation.recipedetail.RecipeDetailActivity
+import com.example.reciperoulette.presentation.recipelist.ui.theme.RecipeRouletteTheme
+
+class MainActivity : ComponentActivity() {
+    @SuppressLint("UnusedMaterial3ScaffoldPaddingParameter")
```
Comment (anandwana001) [inline SachinSinghDighan/RecipeRoulette/app/src/main/java/com/example/reciperoulette/presentation/recipelist/MainActivity.kt]: let's not suppress anything and use the padding param

File: SachinSinghDighan/RecipeRoulette/app/src/main/java/com/example/reciperoulette/presentation/recipelist/MainActivity.kt [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+                    RecipeList(modifier = Modifier.fillMaxSize(), it)
+                }
+            }
+        }
+    }
+}
+
+
+@Composable
+fun RecipeList(modifier: Modifier = Modifier, paddingValues: PaddingValues) {
```
Comment (anandwana001) [inline SachinSinghDighan/RecipeRoulette/app/src/main/java/com/example/reciperoulette/presentation/recipelist/MainActivity.kt]: don't keep compose functions in one file, each function should go to its own file

File: SachinSinghDighan/RecipeRoulette/app/src/main/java/com/example/reciperoulette/presentation/recipedetail/RecipeDetailActivity.kt [hunk 4]
```diff
@@ -0,0 +1,317 @@
...
+import com.example.make_food.ui.commonui.HorizontalListImage
+import com.example.make_food.ui.commonui.HorizontalListTitleText
+import com.example.make_food.ui.commonui.RecipeImage
+import com.example.make_food.ui.commonui.ScreenContentText
+import com.example.make_food.ui.commonui.ScreenSubTitleText
+import com.example.make_food.ui.commonui.ScreenTitleText
+import com.example.reciperoulette.R
+import com.example.reciperoulette.presentation.recipedetail.ui.theme.RecipeRouletteTheme
+
+class RecipeDetailActivity : ComponentActivity() {
```
Comment (anandwana001) [inline SachinSinghDighan/RecipeRoulette/app/src/main/java/com/example/reciperoulette/presentation/recipedetail/RecipeDetailActivity.kt]: why we need 2 activity?

**Review by anandwana001 — CHANGES_REQUESTED** [review]

File: SachinSinghDighan/RecipeRoulette/app/src/main/java/com/example/reciperoulette/presentation/recipelist/MainActivity.kt (hunk 1 above)
Comment (sachinsinghd11) [reply SachinSinghDighan/RecipeRoulette/app/src/main/java/com/example/reciperoulette/presentation/recipelist/MainActivity.kt]: Removed hardcoded strings

File: README.md
Comment (sachinsinghd11) [reply README.md:1]: moved file to my directory

File: SachinSinghDighan/RecipeRoulette/app/src/main/java/com/example/reciperoulette/presentation/recipelist/MainActivity.kt (hunk 2 above)
Comment (sachinsinghd11) [reply SachinSinghDighan/RecipeRoulette/app/src/main/java/com/example/reciperoulette/presentation/recipelist/MainActivity.kt]: removed annotation and used padding

File: SachinSinghDighan/RecipeRoulette/app/src/main/java/com/example/reciperoulette/presentation/recipelist/MainActivity.kt (hunk 3 above)
Comment (sachinsinghd11) [reply SachinSinghDighan/RecipeRoulette/app/src/main/java/com/example/reciperoulette/presentation/recipelist/MainActivity.kt]: created different files for different compose functions

File: SachinSinghDighan/RecipeRoulette/app/src/main/java/com/example/reciperoulette/presentation/recipedetail/RecipeDetailActivity.kt (hunk 4 above)
Comment (sachinsinghd11) [reply SachinSinghDighan/RecipeRoulette/app/src/main/java/com/example/reciperoulette/presentation/recipedetail/RecipeDetailActivity.kt]: Removed 2nd activity and created different composed functions for different screens

File: README.md
Comment (anandwana001) [reply README.md:1]: not moved, still outside

File: README.md [hunk 5]
```diff
@@ -1 +1,16 @@
...
+2) Created Android Project
+3) Implemented Ui Design for -
+   a) Recipe List Screen -
+    ![Recipe_list_screen](https://github.com/user-attachments/assets/17b54678-1806-45b5-a81c-617478825a96)
+    ![Recipe_list _screen2](https://github.com/user-attachments/assets/ba71e703-bc3c-4730-a8fa-21b88457e961)
+
+   b) Recipe Detail Screen -
+    ![Recipe_detail_screen1](https://github.com/user-attachments/assets/6db65b9f-48a6-4a16-8ea0-8af53f11facb)
+    ![Recipe_detail_screen2](https://github.com/user-attachments/assets/8c777acf-eec0-482d-a1ee-3618ce04b027)
+
```
Comment (anandwana001) [inline README.md]: don't add in the root readme, please create your own readme inside your project directory

**Review by anandwana001 — CHANGES_REQUESTED** [review]

File: README.md (hunk 5 above)
Comment (sachinsinghd11) [reply README.md]: Added README to project directory

**Pull request merged**